File: 38. Count and Say.cpp
class Solution {
public:
    string countAndSay(int n) 
    {
        if(n==1) return "1";

        string ans, say = countAndSay(n-1);

        int cnt = 1, nn = say.size();
        char ch = say[0];
        for(int i=1;i<nn;i++)
        {
            if(say[i]==ch) cnt++;
            else 
            {
                ans += to_string(cnt);
                ans += ch;
                cnt = 1;
                ch = say[i];
            }
        }
        ans += to_string(cnt);
        ans += ch;
        return ans;
    }
};

File: 38. Count and Say.py
class Solution:
    def countAndSay(self, n: int) -> str:
        if n == 1:
            return "1"
        s = self.countAndSay(n - 1)

        count, l = 1, len(s)
        convert, prev = "", s[0]
        for i in range(1, l):
            if s[i] == prev: 
                count += 1
            else:
                convert += str(count) + prev
                prev = s[i]
                count = 1
        return convert + str(count) + prev